fireant/dataset/data_blending.py

```diff
@@ -0,0 +1,108 @@
+from fireant import (
+    Field,
+)
+from fireant.dataset.klass import (
+    DataSet,
+)
+from fireant.queries import (
+    DimensionChoicesQueryBuilder,
+    DimensionLatestQueryBuilder,
+)
+from fireant.queries.builder.dataset_blender_query_builder import DataSetBlenderQueryBuilder
+from fireant.utils import immutable
+
+
+class DataSetBlender(object):
+    """
+    WRITEME
+    """
+
+    def __init__(self, primary_dataset):
+        """
+        Constructor for a data blended slicer.  Contains all the fields to initialize the slicer.
+
+        :param primary_dataset: (Required)
+            The primary dataset, which table will be used as part of the FROM expression.
+        """
+        self.primary_dataset = primary_dataset
+        self.table = primary_dataset.table
+        self.secondary_datasets = []
+        self.fields = DataSet.Fields()
+
+        self.field_mapping = {}
+
+        primary_dataset_field_mapping = {}
+
+        for primary_field in self.primary_dataset.fields:
+            primary_dataset_field_mapping[primary_field] = primary_field
+
+        self.field_mapping[primary_dataset.table._table_name] = primary_dataset_field_mapping
```

Comment:
using the name name here is probably not a good idea. Perhaps `str(primary_dataset.table)` to get the fully qualified name and to avoid referencing private attributes?

Reply:
Done. I'm using the table instance now.